src/server/worker/master.go

```diff
@@ -866,19 +895,22 @@ func (a *APIServer) receiveSpout(ctx context.Context, logger *taggedLogger) erro
 					} else {
 						_, err = a.pachClient.PutFile(repo, commit.ID, fileHeader.Name, outTar)
 						if err != nil {
+							fmt.Println("put file error", err)
 							return err
 						}
 					}
 				}
-				// close commit
-				err = a.pachClient.FinishCommit(repo, commit.ID)
```

Comment:
The problem here IIRC was that commits were not always getting finished, which would then cause the whole spout to hang since it couldn't start new commits.